### PR TITLE
Add function to read SBML model from a string

### LINKED ISSUE #111: `parseSBML` function to directly pass in an XML string
it would work identically to readSBML, but the string of the model is in RAM, not saved to disk. It is annoying to have to save then read sometimes

### PR DESCRIPTION
**Pull request by giordano**
Fix #111.  CC: @anandijain

### CODE REVIEW

File: src/readsbml.jl [hunk 1]
```diff
@@ -71,6 +71,34 @@ function get_optional_double(x::VPtr, is_sym, get_sym)::Maybe{Float64}
     end
 end
 
+function _readSBML(
+    symbol::Symbol,
+    fn::String,
+    sbml_conversion = document -> nothing,
+    report_severities = ["Fatal", "Error"],
```
Comment (exaexa) [inline src/readsbml.jl:78]: this probably doesn't need the defaults but I guess it's okay

(is there a good julia way to "forward" the fact that the argument is missing, apart form `Missing` ?)

File: src/readsbml.jl [hunk 2]
```diff
@@ -97,32 +127,37 @@ m = readSBML("my_model.xml", doc -> begin
...
+        report_severities = ["Fatal", "Error"],
+    )::SBML.Model
 
-        if 0 == ccall(sbml(:SBMLDocument_isSetModel), Cint, (VPtr,), doc)
-            throw(AssertionError("SBML document contains no model"))
-        end
+Read the SBML from the string `str` and return the contained `SBML.Model`.
 
-        model = ccall(sbml(:SBMLDocument_getModel), VPtr, (VPtr,), doc)
+The `sbml_conversion` is a function that does an in-place modification of the
```
Comment (exaexa) [inline src/readsbml.jl:145]: avoid copypastes, just refer that "the other arguments work as with `readSBML`"

File: test/loadmodels.jl [hunk 3]
```diff
@@ -58,7 +58,7 @@ sbmlfiles = [
     ),
 ]
 
-@testset "Loading of models from various sources" begin
+@testset "Loading of models from various sources - $(reader)" for reader in (readSBML,  readSBMLFromString)
```
Comment (exaexa) [inline test/loadmodels.jl:61]: lovely.

File: src/readsbml.jl (hunk 2 above)
Comment (giordano) [reply src/readsbml.jl]: Done!

File: src/readsbml.jl (hunk 1 above)
Comment (giordano) [reply src/readsbml.jl:78]: Oh, yeah, this internal function doesn't really need a default, I'm going to remove it.

> (is there a good julia way to "forward" the fact that the argument is missing, apart form Missing ?)

I guess the julian way would be a different method.

**Review by exaexa — APPROVED** [review]

**Pull request merged**